app/stac_api/models.py

```diff
@@ -510,6 +511,24 @@ def delete(self, *args, **kwargs):  # pylint: disable=signature-differs
         super().delete(*args, **kwargs)
 
     def validate_datetime_properties(self):
+        if self.properties_datetime is not None and not isinstance(
+            self.properties_datetime, datetime
+        ):
+            message = "Property datetime must be valid datetime.datetime or None."
+            raise ValidationError(_(message))
+
+        if self.properties_start_datetime is not None and not isinstance(
+            self.properties_start_datetime, datetime
+        ):
+            message = "Property start_datetime must be valid datetime.datetime or None."
+            raise ValidationError(_(message))
+
+        if self.properties_end_datetime is not None and not isinstance(
+            self.properties_end_datetime, datetime
+        ):
+            message = "Property end_datetime must be valid datetime.datetime or None."
+            raise ValidationError(_(message))
```

Comment:
Not a good idea, the fields itself should do the validation, and as it is a DatetimeField it will checks for valid datetime string. This function might be called with valid datetime string instead of datetime object and the transformation happens later. How and where did you had an issue that needed this code ?

Reply:
As discussed I added a check (in a separate function `validate_datetime_string()`) that validates in case the datetime_property is not an instance of datetime if it is a valid datetime string.